File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Sag/AlibabaCloud/SDK/ROS/CDK/Sag/ACLAssociationProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Sag
{
    #pragma warning disable CS8618

    /// <summary>Properties for defining a `ALIYUN::SAG::ACLAssociation`.</summary>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-sag.ACLAssociationProps")]
    public class ACLAssociationProps : AlibabaCloud.SDK.ROS.CDK.Sag.IACLAssociationProps
    {
        /// <summary>Property aclId: Access control ID.</summary>
        [JsiiProperty(name: "aclId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        public object AclId
        {
            get;
            set;
        }

        /// <summary>Property smartAgId: An intelligent gateway instance that needs to bind access control.</summary>
        [JsiiProperty(name: "smartAgId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        public object SmartAgId
        {
            get;
            set;
        }
    }
}

File: multiple-languages/go/alicloudroscdksag/ACLAssociationProps.go
package alicloudroscdksag


// Properties for defining a `ACLAssociation`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/aliyun-sag-aclassociation
type ACLAssociationProps struct {
	// Property aclId: Access control ID.
	AclId interface{} `field:"required" json:"aclId" yaml:"aclId"`
	// Property smartAgId: An intelligent gateway instance that needs to bind access control.
	SmartAgId interface{} `field:"required" json:"smartAgId" yaml:"smartAgId"`
}

